src/cairoUtilFuncGen/utils/encodeToFelt.ts

```diff
@@ -0,0 +1,283 @@
+import assert from 'assert';
+import {
+  ArrayType,
+  BytesType,
+  DataLocation,
+  FunctionCall,
+  generalizeType,
+  getNodeType,
+  SourceUnit,
+  StringType,
+  StructDefinition,
+  typeNameToTypeNode,
+  TypeNode,
+  UserDefinedType,
+} from 'solc-typed-ast';
+import { AST } from '../../ast/ast';
+import { printTypeNode } from '../../utils/astPrinter';
+import {
+  CairoDynArray,
+  CairoStruct,
+  CairoType,
+  TypeConversionContext,
+} from '../../utils/cairoTypeSystem';
+import { NotSupportedYetError } from '../../utils/errors';
+import { createCairoFunctionStub, createCallToFunction } from '../../utils/functionGeneration';
+import { createBytesTypeName } from '../../utils/nodeTemplates';
+import {
+  getElementType,
+  isDynamicArray,
+  isStruct,
+  isValueType,
+} from '../../utils/nodeTypeProcessing';
+import { mapRange, narrowBigIntSafe, typeNameFromTypeNode } from '../../utils/utils';
+import { CairoFunction, delegateBasedOnType, StringIndexedFuncGen } from '../base';
+import { ExternalDynArrayStructConstructor } from '../calldata/externalDynArray/externalDynArrayStructConstructor';
+
+const IMPLICITS = '';
+
+// TODO:
+// What happens with sructs defined in other contract different than the one using new
```

Comment:
Is it worth adding a test for this before merging?

Reply:
Yep, it is bugged when it's a global struct definition and it is used across contracts. Already added ticket for those on the Notion board